flink-streaming-java/src/main/java/org/apache/flink/streaming/util/FieldAccessor.java

```diff
@@ -182,11 +182,14 @@ public R set(R record, F fieldValue) {
 			@SuppressWarnings("unchecked")
 			CompositeType<R> cType = (CompositeType<R>) type;
 
+			if(field.contains(".")) {
+				throw new IllegalArgumentException("The Pojo field accessor currently doesn't support nested POJOs");
```

Comment:
Note that the problem reported in [FLINK-3697](https://issues.apache.org/jira/browse/FLINK-3697) was not related to accessing a nested field.. Perhaps there should be a subsequent JIRA ticket to support nested POJO accessors.